eth/catalyst/api.go

```diff
@@ -281,13 +287,14 @@ func (api *ConsensusAPI) ForkchoiceUpdatedWithWitnessV2(update engine.Forkchoice
 // generates an execution witness too if block building was requested.
 func (api *ConsensusAPI) ForkchoiceUpdatedWithWitnessV3(update engine.ForkchoiceStateV1, params *engine.PayloadAttributes) (engine.ForkChoiceResponse, error) {
 	if params != nil {
-		switch {
-		case params.Withdrawals == nil:
-			return engine.STATUS_INVALID, attributesErr("missing withdrawals")
-		case params.BeaconRoot == nil:
-			return engine.STATUS_INVALID, attributesErr("missing beacon root")
-		case !api.checkFork(params.Timestamp, forks.Cancun, forks.Prague):
-			return engine.STATUS_INVALID, unsupportedForkErr("fcuV3 must only be called for cancun or prague payloads")
+		if params.Withdrawals == nil {
+			return engine.STATUS_INVALID, engine.InvalidPayloadAttributes.With(errors.New("missing withdrawals"))
```

Comment:
The latest rebase removed the refactors that were recently merged into this file.

Reply:
Current implementation expects fcuv3 be called in eip7805 fork as well, which will be absorbed into some canonical fork in future.

Reply:
The review wasn't about this line in particular, sorry for the confusion.  We refactored a lot of `if`/`else if`/`else if`/... into switch statements.

Reply:
Oh right, the last commit reverts all of those changes without causing any conflict. My bad, let me reflect the refactors.